apis/clusterresources/v1beta1/opensearchegressrules_webhook.go

```diff
@@ -65,6 +65,11 @@ func (r *OpenSearchEgressRules) ValidateCreate() error {
 		return fmt.Errorf("the type should be equal to one of the options: %q , got: %q", destinationTypes, r.Spec.Type)
 	}
 
+	if (r.Spec.ClusterID == "" && r.Spec.ClusterRef == nil) ||
+		(r.Spec.ClusterID != "" && r.Spec.ClusterRef != nil) {
+		return fmt.Errorf("only one of dataCenter ID and cluster reference fields should be specified")
+	}
```

Comment:
Here you don't check for data center ID, please change error message

Reply:
fixed